src/compiler/binder.ts

```diff
@@ -3807,7 +3807,7 @@ namespace ts {
         }
 
         // Type annotations are TypeScript syntax.
-        if (node.type) {
+        if (node.type || node.exclamationToken) {

```

Comment:
is this change related to the emitter?
If so, you probably also want to make the same change in `computePropertyDeclaration` (see #35097 which I erroneously opened)